### ISSUE TITLE
Update for .NET 8.0, consider removing deprecated .NET versions

### ISSUE THREAD

**Issue by JohnCampionJr** [toddams/RazorLight#538]
It would be good to update this to directly support .net8.0.

Should also consider removing support for  pre .net6.0 versions (core 3.1, net5.0)

Happy to submit a PR to do this, but wanted to check first.

**Comment by BC89** [comment]
Not that .net8 is LTS this would be awesome.

**Comment by BC89** [comment]
I was looking this over and seems like the dependency on System.Drawing.Common is a non-starter if you are deploying to a Linux box. In my case, I have a Linux Docker container running .Net 8 and that seems to spell the end of the line for me.  From what I can tell .Net 6 was the last Linux version that included the necessary code bits -https://github.com/dotnet/runtime/issues/80148

`System.Drawing.Common is not supported on non-Windows platforms. See https://aka.ms/systemdrawingnonwindows for more information.endencey  on System.Drawing.Common is not supported on non-Windows platforms. See https://aka.ms/systemdrawingnonwindows for more information.`

**Comment by A9G-Data-Droid** [comment]
NET8 brings [a new way to render razor pages](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-components-outside-of-aspnetcore?preserve-view=true&view=aspnetcore-8.0)

So this could result in a major back end overhaul to a new Microsoft supported workflow.

**Comment by blackholeearth** [comment]
İ hope thşs doesn't remove .net 4.6 2 support.

Win7 forever....

**Comment by GaryWayneSmith** [comment]
> İ hope thşs doesn't remove .net 4.6 2 support.
> 
> Win7 forever....

We have a lot of legacy apps using .Net 4.6.2.  We are migrating libs over to dual .Net Framework 4.6.2/Standard 2.1 so we can do new projects with some shared functionality.  I'm using the old RazorEngine that's been dead for a while.  A lot of open-source developers seem dumping old .Net framework stuff which I think is bad for legacy apps that have been stable for years.

I have two libraries that I can no longer update because the next nuget update is incompatible (for no reason at all as I can download their source, just change it to Standard 2.0, and recompile (with maybe a minor conditional change).

I hope this project doesn't go that route as we're starting to use this as a replacement for the former engine.

**Comment by A9G-Data-Droid** [comment]
> İ hope thşs doesn't remove .net 4.6 2 support.
> 
> Win7 forever....

I'm not sure why you would use 4.6 when 4.8, .NET5, and .NET6 were all supported on Windows 7. The OS itself is out of support and yet it still works. The same can be said for running .NET8 on Windows 7. It isn't supported but it might work. Really depends on what libraries you end up using.

https://github.com/dotnet/core/issues/7556

**Comment by molinch** [comment]
@JohnCampionJr are there workarounds needed to run on . net8

**Comment by jzabroski** [comment]
RazorLight should no longer really be needed now that NET8 brings [a new way to render razor pages](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-components-outside-of-aspnetcore?preserve-view=true&view=aspnetcore-8.0), as @A9G-Data-Droid noted.

**Comment by jzabroski** [comment]
What would you all still use RazorLight for?

**Comment by ktb-dev** [comment]
>  NET8 brings [a new way to render razor pages](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-components-outside-of-aspnetcore?preserve-view=true&view=aspnetcore-8.0)

@A9G-Data-Droid , @jzabroski 
could you tell me how to render a template obtained from the database using a new way?

**Comment by jzabroski** [comment]
RazorLightEngine would just call the HtmlRenderer in net8.0 in an InvokeAsync context, as the linked sample suggests. Open to a PR that uses

```c#
#if NET8_0_OR_GREATER

#endif
```

Some review of caching may be required as Im not sure how the new approach by Microsoft disposes of compiled resources if the template changes a lot.

**Comment by nm-a** [comment]
Hi @jzabroski 

I feel it is better to have new branch which targets >= .NET 8 with new nuget package version

**Comment by betsu-nega** [comment]
> RazorLightEngine would just call the HtmlRenderer in net8.0 in an InvokeAsync context, as the linked sample suggests. Open to a PR that uses
> 
> ```cs
> #if NET8_0_OR_GREATER
> 
> #endif
> ```
> 
> Some review of caching may be required as Im not sure how the new approach by Microsoft disposes of compiled resources if the template changes a lot.

I think the new way only supports rendering 'Razor Components'. It doesn't support rendering 'html' or 'string' templates.

**Comment by michaelakin** [comment]
> I think the new way only supports rendering 'Razor Components'. It doesn't support rendering 'html' or 'string' templates.


Yes, this is what I am struggling with right now.  Wish HtmlRenderer  would work like this packages examples work. 

is this package still supported?

**Comment by PaulIvanov** [comment]
Is there any prioritization on updating to support .NET8's new rendering patterns? Or has someone figured out the best path forward to continue using this library in .NET8?

**Comment by alekdavis** [comment]
@PaulIvanov I am using this library with .NET 8 with no problems. Not sure about the new rendering patterns, though, but I use it for generating HTML email from static template files and it works fine.

**Comment by PaulIvanov** [comment]
Thanks @alekdavis for confirmation. We are doing a technical audit of a large codebase that is in .Net Core 3.1 utilizing this library. I thought this issue was stating that it no longer works in .NET 8. Good to know we can continue to rely on it as we plan to upgrade to .Net 8

**Comment by alekdavis** [comment]
@PaulIvanov Yes, the only issue is that you need to modify the project settings as oulined at https://github.com/toddams/RazorLight?tab=readme-ov-file#im-getting-cant-load-metadata-reference-from-the-entry-assembly-exception (not sure why we need this, since we're using external template files, but it's not a big deal).

**Comment by JohnCampionJr** [comment]
I have made and published a fork of this project here: https://github.com/jcamp-code/RazorLight

I'm updating my FluentEmail packages to use it.
